### PR TITLE
feat(framework) Add `flwr install` command

### PR DESCRIPTION
**Pull request by charlesbvll**
### Checklist

- [x] Implement proposed change
- [x] Make CI checks pass
- [x] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/py/flwr/cli/install.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+) -> None:
+    """Install a Flower project from a FAB file.
+
+    It can be ran with a single FAB file argument:
+
+        ``flwr install ./docs/target_project.fab``
+
+    The target install directory can be specified with ``--flwr-dir``:
+
+        ``flwr install ./docs/target_project.fab --flwr-dir ./docs/flwr``
```
Comment (danieljanes) [inline src/py/flwr/cli/install.py:50]:
```suggestion
        ``flwr install ./target_project.fab``

    The target install directory can be specified with ``--flwr-dir``:

        ``flwr install ./target_project.fab --flwr-dir ./docs/flwr``
```

File: src/py/flwr/cli/install.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    source: Annotated[
+        Optional[Path],
+        typer.Argument(metavar="source", help="The source FAB file to install."),
+    ] = None,
+    flwr_dir: Annotated[
+        Optional[Path],
+        typer.Option(help="The desired install path."),
+    ] = None,
+) -> None:
+    """Install a Flower project from a FAB file.
```
Comment (danieljanes) [inline src/py/flwr/cli/install.py:42]:
```suggestion
    """Install a Flower App Bundle.
```

File: src/py/flwr/cli/install.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    if config is None:
+        typer.secho(
+            "❌ Invalid config inside FAB file.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    username = config["flower"]["publisher"]
```
Comment (danieljanes) [inline src/py/flwr/cli/install.py:133]: Let's call this `publisher` consistently:
```suggestion
    publisher = config["flower"]["publisher"]
```

File: src/py/flwr/cli/install.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    project_name = config["project"]["name"]
+    version = config["project"]["version"]
+
+    if fab_name != f"{username}.{project_name}.{version.replace('.', '-')}":
+        typer.secho(
+            "❌ FAB file has incorrect name, it should be "
+            "`<username>.<project_name>.<version>.fab`.",
```
Comment (danieljanes) [inline src/py/flwr/cli/install.py:140]:
```suggestion
            "❌ FAB file has incorrect name. The file name must follow the format "
            "`<username>.<project_name>.<version>.fab`.",
```

File: src/py/flwr/cli/install.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+def _get_sha256_hash(file_path: Path) -> str:
+    """Calculate the SHA-256 hash of a file."""
+    sha256 = hashlib.sha256()
+    with open(file_path, "rb") as f:
+        while True:
+            data = f.read(65536)
+            if not data:
+                break
+            sha256.update(data)
+    return sha256.hexdigest()
```
Comment (danieljanes) [inline src/py/flwr/cli/install.py:208]: We have a similar function in `flwr.cli.build`, can we merge them? We can move it to `utils`

File: src/py/flwr/cli/install.py (hunk 4 above)
Comment (charlesbvll) [reply src/py/flwr/cli/install.py:140]: @danieljanes I was planning to remove this check in the future as it is not relevant for when we are working with raw bytes and also I think it should be possible to install the FAB even though the filename is something else. Wdyt? I can also add a new parameter like `check_filename` that would be set to `True` when using the CLI

**Review by danieljanes — APPROVED** [review]

**Pull request merged**